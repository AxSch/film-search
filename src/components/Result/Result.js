import React from 'react'
import { Link } from 'react-router-dom'
import { ResultItem, ResultName } from './Result.styled'

const Result = ({ resultData, category }) => {
    const handleCategory = () => {
        if (category === 'movies') {
            return (
                <ResultItem data-testid="resultItemMovies">
                    <Link to={`/item/${resultData.id}&category=${category}`}>
                        <span>{resultData.title}</span>
                    </Link>
                </ResultItem>
            )
        } else {
            return (
                <ResultItem data-testid="resultItem">
                    <Link to={`/item/${resultData.id}&category=${category}`}>
                        <ResultName data-testid="resultName">{resultData.name}</ResultName>
                        {resultData.known_for_department && resultData.known_for_department !== '' ? <span>- {resultData.known_for_department}</span> : null}
                    </Link>
                </ResultItem>
            )
        }
    }

    return (
        <>
            {handleCategory()}
        </>
    )
}

export default Result
