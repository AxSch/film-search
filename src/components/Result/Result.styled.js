import styled from 'styled-components'

const ResultItem = styled.li`
    margin-bottom: 0.5rem;

    a {
        color: white;
        text-decoration: none;
        &:hover {
            color: rgba(255,255,255, 0.4);
        }
    }
`

const ResultName = styled.span`
    margin-right: 0.4rem;
`

export {
    ResultItem,
    ResultName
}
