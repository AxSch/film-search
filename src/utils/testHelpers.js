import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom'
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import searchResultReducer from '../reducers/searchResults/searchResultSlice';

export const renderWithRouter = (ui, { route = '/' } = {}) => {
    window.history.pushState({}, 'Test page', route)
  
    return render(ui, { wrapper: BrowserRouter })
  }


export const renderConnected = (
  ui, {
    store = createStore(searchResultReducer, {}),
    route = '/',
    ...renderOptions
  } = {}
) => {
  window.history.pushState({}, 'Test page', route)
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...renderOptions});
};

export default renderConnected;