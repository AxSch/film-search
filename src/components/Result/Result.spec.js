import React from 'react'
import { renderWithRouter } from '../../utils/testHelpers';
import Result from './Result'

describe('Result', () => {
    let container, getByTestId, initialProps
    beforeEach(() => {
        initialProps = {
            category: 'tvShows',
            resultData: {
                adult: false,
                gender: 0,
                id: 2400560,
                known_for: [{}],
                known_for_department: "Acting",
                name: "Willie Smith",
                popularity: 0.6,
                profile_path: null
            }
        }
        const utils = renderWithRouter(<Result resultData={initialProps.resultData} category={initialProps.category} />);
        container = utils.container;
        getByTestId = utils.getByTestId;
    })

    it('should render successfully', () => {
        expect(getByTestId("resultItem")).toBeInTheDocument();
        expect(getByTestId("resultName")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('should render movies result', () => {
        initialProps.category = "movies" 
        initialProps.resultData = {
            title: 'Monsoon Lagoon',
            id: 2400
        }
        const utils = renderWithRouter(<Result resultData={initialProps.resultData} category={initialProps.category} />);
        container = utils.container;
        getByTestId = utils.getByTestId;
        expect(getByTestId("resultItemMovies")).toBeInTheDocument();
    })

})