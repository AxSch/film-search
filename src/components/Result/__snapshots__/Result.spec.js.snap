// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render successfully 1`] = `
<div>
  <li
    class="sc-bdnxRM crZWod"
    data-testid="resultItem"
  >
    <a
      href="/item/2400560&category=tvShows"
    >
      <span
        class="sc-gtsrHT kiexgk"
        data-testid="resultName"
      >
        Willie Smith
      </span>
      <span>
        - 
        Acting
      </span>
    </a>
  </li>
</div>
`;
